staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -411,9 +411,15 @@ func (r *Request) Body(obj interface{}) *Request {
 	return r
 }
 
-// Context adds a context to the request. Contexts are only used for
+// Context returns the context currently associated with the request. To change
```

Comment:
We generally don’t expose readers for things on request, so I’m not sure we should add this without a use case

Reply:
Maybe leave Context with the old definition so we don't break compatibility?

Reply:
Is the current signature of Context something that we could deprecate? If no, then I think one API is better than two, even if it's inconsistent with convention.